go/service/main.go

```diff
@@ -590,6 +593,24 @@ func (d *Service) runMerkleAudit(ctx context.Context) {
 	d.G().PushShutdownHook(eng.Shutdown)
 }
 
+func (d *Service) runInitializeInviteFriendsCounts(ctx context.Context) {
+	go d.runInitializeInviteFriendsCountsInner(ctx)
+}
+func (d *Service) runInitializeInviteFriendsCountsInner(ctx context.Context) {
+	mctx := libkb.NewMetaContext(ctx, d.G())
+
+	if !mctx.G().UIRouter.WaitForUIType(libkb.HomeUIKind, 30*time.Second) {
```

Comment:
I'd drop a TODO here (and a ticket?) to improve this function. My typical workflow would break it:

1. Start `yarn start-hot`, `yarn hot-server` and `go install -v ... && keybase --debug service` in three tabs.
2. `start-hot` finishes first and goes into the infinite loop waiting for `hot-server`.
3. go compilation finishes second and the service gets started
4. webpack is still looking for a purpose in its life
5. webpack is _still_ not done
6. webpack (hot-server) starts, start-hot opens the window, but 30s already passed so the number is not present in the frontend

A simple fix would be to move the counter into a variable and hook into registering HomeUI, send an update as soon as it connects.

Reply:
good point, i also didn't want to send off that api call immediately on service start though. want to make a ticket for that change?

Reply:
I'd appreciate if you could file it, I'll keep this tab pinned so hopefully I won't miss it.